tests/integration/issues/github_1229/test_sharding_empty_index.py

```diff
@@ -0,0 +1,63 @@
+import os
+
+import numpy as np
+
+from jina import Flow, Document
+
+callback_was_called = False
+
+
+def get_flow():
+    num_shards = 2
+    f = Flow() \
+        .add(
+        uses='vectorindexer.yml',
+        shards=num_shards,
+        separated_workspace=True,
+        uses_after='_merge_all',
+        polling='all'
+    )
+    return f
+
+
+def test_sharding_empty_index(tmpdir):
+    print(f'WORKSPACE = {tmpdir}')
+    os.environ['JINA_TEST_1229_WORKSPACE'] = os.path.abspath(tmpdir)
+
+    f = get_flow()
+
+    num_docs = 1
+    data = []
+    for i in range(num_docs):
+        with Document() as doc:
+            doc.content = f'data {i}'
+            doc.embedding = np.array([i])
+            data.append(doc)
+
+    with f:
+        f.index(data)
+
+    f.close()
+
+    f = get_flow()
+
+    num_query = 10
+    query = []
+    for i in range(num_query):
+        with Document() as doc:
+            doc.content = f'query {i}'
+            doc.embedding = np.array([i])
+            query.append(doc)
+
+    def callback(result):
+        global callback_was_called
```

Comment:
This is something that I find interesting:
1. On one side I hate to see the `global` variable, but I like this idea that guarantees that we do not have false positivies

However, 

we could pass to the callback some `container` to be switched. (We could wrap a boolean in a class that it can be switched in the callback). This could even be made a `fixture` available for every test (which may not be available for the global variable)